src/lib/polymorph.tsx

```diff
@@ -1,17 +1,46 @@
 import * as React from "react";
-import { omit } from "onno-react";
+import {
+  omit,
+  spaceSet,
+  backgroundSet,
+  borderSet,
+  colorSet,
+  displaySet,
+  flexSet,
+  gridSet,
+  layoutSet,
+  outlineSet,
+  textSet,
+  transformSet,
+} from "onno-react";
 
 export interface PolymorphProps {
   as?: React.ElementType;
 }
 
-const omitPolymorphProps = omit({ propsKeys: ["as"] });
+const omitStyleProps = omit({
+  propsKeys: [
+    "as",
+    ...backgroundSet.options.propsKeys,
+    ...borderSet.options.propsKeys,
+    ...colorSet.options.propsKeys,
+    ...displaySet.options.propsKeys,
+    ...flexSet.options.propsKeys,
+    ...gridSet.options.propsKeys,
+    ...layoutSet.options.propsKeys,
+    ...outlineSet.options.propsKeys,
+    ...spaceSet.options.propsKeys,
+    ...textSet.options.propsKeys,
+    ...transformSet.options.propsKeys,
+  ],
+});
 
 export function polymorph<P extends unknown>(
   defaultEl: React.ElementType,
 ): React.FunctionComponent<PolymorphProps & P> {
   return props => {
     const Element = props.as || defaultEl;
-    return <Element {...omitPolymorphProps(props)} />;
+    return <Element {...omitStyleProps(props)} />;
```

Comment:
What about just `omitProps`?

Reply:
I find `omitStyleProps` to be more descriptive, but am open to the change if you feel it is necessary.